### PR TITLE
Git utilities

### PR DESCRIPTION
**Pull request by kewegner**
the function GetGitInfo returns a struct with the following:

repo URL
branch
Boolean whether changes have been made

Last commit Info:
 commit long SHA
 commit author
 commit datetime
 commit URL

This code has only been tested in a local workspace environment, and would need to be tested in the pipeline environments, and with a wider set of git project environments.

### CODE REVIEW

File: cmd/git_utils.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		}
+
+	} else {
+		Info.log("Unable to determine origin to compute repository URL")
+	}
+
+	gitInfo.Commit, gitErr = RunGitGetLastCommit(gitInfo.RemoteURL, dryrun)
+	if gitErr != nil {
+		Info.log("Received error getting current commit: ", gitErr)
+		//return gitInfo, gitErr
```
Comment (EnriqueL8) [inline cmd/git_utils.go]: Should we return an error, or return the `gitInfo` struct without the current commit? I am not sure

File: cmd/git_utils.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	lineSeparator := "\n"
+	if runtime.GOOS == "windows" {
+		lineSeparator = "\r\n"
+	}
+	outputLines := strings.Split(output, lineSeparator)
+
+	var changeLines []string
+	const noCommits = "## No commits yet on "
+	const branchPrefix = "## "
+	const branchSeparatorString = "..."
```
Comment (EnriqueL8) [inline cmd/git_utils.go]: I tested this command locally, and did not get a `branchSeparatorString` after my branch name. 
Instead I get just 
image
So the branch field in GitInfo is not set correctly. Is the separator needed when you have multiple branches?

(Note: git.go is the local version of this file)

File: cmd/git_utils.go (hunk 1 above)
Comment (kewegner) [reply cmd/git_utils.go]: The thought here was not to error out, if we can't find the commit.

File: cmd/git_utils.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	for _, value := range outputLines {
```
[the remark this answers is not in the document]
Comment (kewegner) [reply cmd/git_utils.go]: Kyle and Neeraj had said we only needed a boolean whether there were changes, so I am going to get rid of the changes list for the moment.

File: cmd/git_utils.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+			}
+			// this must be a change line
+			if !branchFound {
+				changeLines = append(changeLines, value)
+			}
+		}
+
+	}
+	gitInfo.Changes = append(gitInfo.Changes, changeLines...)
```
[the remark this answers is not in the document]
Comment (kewegner) [reply cmd/git_utils.go]: See above.

File: cmd/git_utils.go [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+	gitInfo.Commit, gitErr = RunGitGetLastCommit(gitInfo.RemoteURL, dryrun)
+	if gitErr != nil {
+		Info.log("Received error getting current commit: ", gitErr)
+		//return gitInfo, gitErr
+	}
+	changesMade := false
+	if len(changeLines) != 0 {
+		changesMade = true
+	}
+	gitInfo.ChangesMade = changesMade
```
[the remark this answers is not in the document]
Comment (kewegner) [reply cmd/git_utils.go]: logic will be change per above.

File: cmd/git_utils.go (hunk 2 above)
Comment (kewegner) [reply cmd/git_utils.go]: what do you get when you do git remote -v out of curiosity?  What about git config?

Comment (kewegner) [reply cmd/git_utils.go]: It's ok I can reproduce it.

**Pull request merged**